TestReadiness/Ready_-_returns_200 flaky

```
{"time":"2024-10-15T18:26:35.847390637Z","level":"ERROR","msg":"error retrieving value from context, value obtained was '<nil>' and type obtained was '*slog.Logger'"}
--- FAIL: TestReadiness (0.00s)
    --- FAIL: TestReadiness/Ready_-_returns_200 (0.00s)
        frontend_test.go:57: expected status code 200, got 500
2024/10/15 18:26:35 http: superfluous response.WriteHeader call from github.com/Azure/ARO-HCP/internal/api/arm.WriteCloudError (error.go:110)
2024/10/15 18:26:35 http: superfluous response.WriteHeader call from github.com/Azure/ARO-HCP/internal/api/arm.WriteCloudError (error.go:110)
2024/10/15 18:26:35 http: superfluous response.WriteHeader call from github.com/Azure/ARO-HCP/internal/api/arm.WriteCloudError (error.go:110)
2024/10/15 18:26:35 http: superfluous response.WriteHeader call from github.com/Azure/ARO-HCP/internal/api/arm.WriteCloudError (error.go:110)
2024/10/15 18:26:35 http: superfluous response.WriteHeader call from github.com/Azure/ARO-HCP/internal/api/arm.WriteCloudError (error.go:110)
2024/10/15 18:26:35 http: superfluous response.WriteHeader call from github.com/Azure/ARO-HCP/internal/api/arm.WriteCloudError (error.go:110)
2024/10/15 18:26:35 http: superfluous response.WriteHeader call from github.com/Azure/ARO-HCP/internal/api/arm.WriteCloudError (error.go:110)
2024/10/15 18:26:35 WARN Failed to parse X-Ms-Arm-Resource-System-Data header: <nil>
FAIL
```

--- a/frontend/pkg/frontend/metrics.go
+++ b/frontend/pkg/frontend/metrics.go
@@ -51,7 +51,7 @@ func (pe *PrometheusEmitter) EmitCounter(name string, value float64, labels map[
 	if !exists {
 		labelKeys := maps.Keys(labels)
 		vec = prometheus.NewCounterVec(prometheus.CounterOpts{Name: name}, labelKeys)
-		prometheus.MustRegister(vec)
+		pe.registry.MustRegister(vec)
 		pe.counters[name] = vec
 	}
 	vec.With(labels).Add(value)

